core/src/com/unciv/logic/automation/BattleHelper.kt

```diff
@@ -102,6 +102,12 @@ object BattleHelper {
         if (tileCombatant.getCivInfo() == combatant.getCivInfo()) return false
         if (!combatant.getCivInfo().isAtWarWith(tileCombatant.getCivInfo())) return false
 
+        //don't let non-embarked melee units attack water units, unless they have Optics tech
+        if (combatant is MapUnitCombatant && !combatant.unit.baseUnit.unitType.contains("Water") &&
+            combatant.unit.baseUnit.isMelee() && !combatant.unit.isEmbarked() && !combatant.getCivInfo().hasTechOrPolicy("Optics") &&
```

Comment:
I'm still not sure if land melee units can attack ships directly from land even after optics in civ 5, but if it is you should probably replace ```!combatant.getCivInfo().hasTechOrPolicy("Optics")``` with ```combatant.unit.civInfo.hasUnique(LandUnitEmbarkation)``` since it would allow mods that change the name of the optics tech or move the ability to a different tech to still apply.

Reply:
I did, I also removed some of the checks since I don't  think they were really necessary

Reply:
yeah that wiki also says land units can attack water units, but doesn't say anything about requiring Optics. Should I remove that check?

Reply:
From the page I linked
> Any embarked land units are NOT considered naval units. As such, they are very vulnerable to attacks, although they can absorb a hit or two. They can't attack other naval units, although they return fire in melee combat.

I haven't tried it out in Civ 5 but it sounds like that embarked units can't initiate combat against a naval target at all, only being able to inflict damage in defense if attacked by a naval melee unit.

You can probably simplify the conditions by just looking at whether the combatant is embarked, whether the tile contains a city, and checking if the target tile ```isWater``` is the same as the combatant tile ```isWater``` since
* land melee units regardless of embarked status cannot attack water units
* water melee units can only attack cities not non-embarked land units
* embarked melee units can attack land

Reply:
That is correct. It was a distinct choice that Melee units should be able to attack Water units in UnCiv, which is different than Civ5